static/scripts/rewards/render-transaction/insert-table-data.ts

```diff
@@ -2,15 +2,51 @@ import { ERC20Permit, ERC721Permit } from "@ubiquibot/permit-generation/types";
 import { BigNumber, ethers } from "ethers";
 import { app, AppState } from "../app-state";
 
+// dynamically shortens a string by slicing and hiding the middle part
 function shortenAddress(address: string): string {
-  return `${address.slice(0, 10)}...${address.slice(-8)}`;
+  const initialLength = 42; // address has 42 chars
+  const maxWidth = 570; // width to trigger shortening
+
+  if (window.innerWidth >= maxWidth) {
+    return address;
+  }
+
+  // remove 1 letter for every 6px below 520px
```

Comment:
Please use pure CSS. I implemented this logic in work.ubq.fi for the issue titles and repository names. You can probably use the same logic.